app/src/test/java/com/globant/pokemontcgapp/pokemontypeviewmodeltest/PokemonTypeViewModelTest.kt

```diff
@@ -0,0 +1,68 @@
+package com.globant.pokemontcgapp.pokemontypeviewmodeltest
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.globant.pokemontcgapp.MockedPokemonTypes
+import com.globant.pokemontcgapp.testObserver
+import com.globant.pokemontcgapp.util.PokemonTypeState
+import com.globant.pokemontcgapp.viewmodel.PokemonTypeViewModel
+import com.globant.pokemontcgapp.viewmodel.contract.PokemonTypeContract
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class PokemonTypeViewModelTest {
+
+    @ObsoleteCoroutinesApi
+    private var mainThreadSurrogate = newSingleThreadContext(UI_THREAD)
```

Comment:
I have a question, Can you test using another thread?

Reply:
yes it can, changed the name of the thread so it's no longer misleading, thanks